sender/splunk.go

```diff
@@ -0,0 +1,174 @@
+/*
+ * skogul, splunk writer
+ *
+ * Copyright (c) 2020 Telenor Norge AS
+ * Author(s):
+ *  - Håkon Solbjørg <Hakon.Solbjorg@telenor.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+package sender
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/telenornms/skogul"
+)
+
+var splunkLog = skogul.Logger("sender", "splunk")
+
+// Splunk contains the configuration parameters for this sender.
+type Splunk struct {
+	URL           string `doc:"URL to Splunk HTTP Event Collector (HEC)"`
+	Token         string `doc:"Token for HTTP Authorization header for HEC endpoint."`
+	Index         string `doc:"Custom Splunk index to send event to."`
+	HostnameField string `doc:"Name of the metadata field with the hostname. Note, this might have to be transformed into metadata depending on the input data."`
+	HTTP          *HTTP  `doc:"HTTP sender options. URL is overwritten from this config, the rest will be HTTP sender defaults unless overridden."`
```

Comment:
The documentation should probably state that this uses the config from the HTTP sender. I'm starting to think we might need/want a richer way to do documentation for certain modules. Not for the individual options, since that would bloat it, but a better way to provide examples/prose of a generic nature. Not relevant for this PR though, just a brain dump.